pcsx2/Recording/InputRecording.cpp

```diff
@@ -63,28 +63,51 @@ void SaveStateBase::InputRecordingFreeze()
 
 InputRecording g_InputRecording;
 
-InputRecording::InputRecording()
+InputRecording::~InputRecording()
 {
-	// NOTE - No multi-tap support, only two controllers
-	padData[CONTROLLER_PORT_ONE] = new PadData();
-	padData[CONTROLLER_PORT_TWO] = new PadData();
+	// As to not attempt a double delete from WX
+	newRecordingFrame.release();
+	openFileDialog.release();
 }
 
-void InputRecording::setVirtualPadPtr(VirtualPad* ptr, int const port)
+InputRecording::InputRecordingPad::InputRecordingPad()
 {
-	virtualPads[port] = ptr;
+	padData = std::make_unique<PadData>();
 }
 
-void InputRecording::RecordingReset()
+InputRecording::InputRecordingPad::~InputRecordingPad()
+{
+	// As to not attempt a double delete from WX
+	virtualPad.release();
+}
+
+void InputRecording::InitInputRecordingWindows(wxWindow* parent)
+{
+	if (newRecordingFrame.get() != nullptr)
+		return;
```

Comment:
this fundamentally works, but seems like a bit of an anti-pattern = using the initialization of 1/4 of the preceding variables as a check for all of them.  

I'd much rather see checking before initializing each, or have an all-encompassing `bool` to check if Recording GUI elements have already been initialized.